pkg/mutator/handler.go

```diff
@@ -0,0 +1,125 @@
+package mutator
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+
+	"github.com/giantswarm/microerror"
+	"k8s.io/api/admission/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	metav1beta1 "k8s.io/apimachinery/pkg/apis/meta/v1beta1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/serializer"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+type Mutator interface {
+	Log(keyVals ...interface{})
+	Mutate(ctx context.Context, review *v1beta1.AdmissionRequest) ([]PatchOperation, error)
+	Resource() string
+}
+
+var (
+	scheme        = runtime.NewScheme()
+	codecs        = serializer.NewCodecFactory(scheme)
+	Deserializer  = codecs.UniversalDeserializer()
+	InternalError = errors.New("internal admission controller error")
+)
+
+func Handler(mutator Mutator) http.HandlerFunc {
```

Comment:
I'm surprised we have to write this code ourselves. I expected kubernetes to provide some skaffolding for this.